src/lua/zencode_w3c.lua

```diff
@@ -145,85 +145,228 @@ ZEN:add_schema(
-    zencode_assert(cred.proof, 'The object is not signed: ' .. vc)
-    local m = have(meth)
-    ACK[vc].proof.verificationMethod = m
-end)
-
-When("get verification method in ''", function(vc)
-    empty 'verification_method'
-    local cred = have(vc)
-    zencode_assert(cred.proof, 'The object is not signed: ' .. vc)
-    ACK.verification_method = cred.proof.verificationMethod
-    new_codec('verification_method')
-end)
-
+-- utlity to encode a JSON as octet string based on its original encoding
+-- if octet are passed as input it will check that the original string was
+-- a json or a json encoded in url64
+-- @param src the json to encode
+-- @return octet string containg the encoded json
 local function _json_encoding(src)
     local source, codec = have(src)
     local source_str = source
     if luatype(source) == 'table' then
         local encoding = codec.schema or codec.encoding
 		   or CODEC.output.encoding.name
         source_str = O.from_string( JSON.encode(source, encoding) )
+    else
+        -- check that before encoding it was a table
+        -- this should ensure that we are signing only json
+        local input = get_encoding_function(codec.encoding)(source)
+        if O.is_url64(input) then
+            input = O.from_url64(input):string()
+        end
+        if not JSON.validate(input) then
+            error(src..' is not a json or an encoded json')
+        end
     end
     return source_str
 end
 
-When("create jws signature of ''", function(src)
-    local source_str = _json_encoding(src)
-    empty'jws'
-    local sk = havekey'ecdh' -- assuming secp256k1
-    ACK.jws = O.from_string(
-        jws_signature_to_octet(ECDH.sign(sk, source_str)) )
-    new_codec('jws', { zentype = 'e',
-                        encoding = 'string' })
+When(deprecated("create jws signature of ''",
+                "create jws detached signature with header '' and payload ''",
+                function(src)
+                    warn('raw signature of the payload, non standard jws')
+                    local source_str = _json_encoding(src)
+                    empty'jws'
+                    local sk = havekey'ecdh' -- assuming secp256k1
+                    ACK.jws = O.from_string(
+                        jws_signature_to_octet(ECDH.sign(sk, source_str)) )
+                    new_codec('jws', { zentype = 'e',
+                                       encoding = 'string' })
+               end)
+)
+
+When(deprecated("create jws signature using ecdh signature in ''",
+                "create jws detached signature with header '' and payload ''",
+                function(sign)
+                    warn('external signature can be not jws compliant')
+                    local signature = have(sign)
+                    empty'jws'
+                    ACK.jws = O.from_string(jws_signature_to_octet(signature))
+                    new_codec('jws', { zentype = 'e',
+                                       encoding = 'string' })
+end))
+
```

Comment:
add stataments to create jws header like:
* create jws header for es256 signature 
* create jws header for es256 signature with public key (if es256 key present, compute pk and add to header as jwk)

both for es256/secp256r1/p256 and ecdh/es256k/secp256k1